n2survey/lime/survey.py

```diff
@@ -635,3 +635,113 @@ def get_choices(self, question: str) -> dict:
             choices_dict = question_info.choices[0]
 
         return choices_dict
+
+    def rate_mental_health(
+        self, question: str, condition: str, attach: bool = False
+    ) -> pd.DataFrame:
+        """Calculate State/Trait Anxiety or Depression score based on responses to question.
+
+        Args:
+            question (str): Question ID to use for calculation
+            condition (str): Which kind of mental health condition to rate, "state", "trait", or "depression"
+            attach (bool): Whether to attach the result DataFrame to the responses DataFrame. Default: False
+
+        Returns:
+            pd.DataFrame: Mental health condition ratings and classifications
+        """
+
+        # set up condition-specific parameters
+        if condition == "state":
+            if "I feel calm" not in self.get_label(question + "_SQ001"):
+                raise ValueError("Question incompatible with specified condition type.")
+            base_score = 10 / 3
+            conversion = ["pos", "neg", "neg", "pos", "pos", "neg"]
+            label = "state_anxiety"
+            classification_boundaries = [0, 37, 44, 80]
+            classes = ["no or low anxiety", "moderate anxiety", "high anxiety"]
+        elif condition == "trait":
+            if "calm, cool and collected" not in self.get_label(question + "_SQ001"):
+                raise ValueError("Question incompatible with specified condition type.")
+            base_score = 5 / 2
+            conversion = [
+                "pos",
+                "neg",
+                "neg",
+                "pos",
+                "neg",
+                "neg",
+                "pos",
+                "neg",
+            ]
+            label = "trait_anxiety"
+            classification_boundaries = [0, 37, 44, 80]
+            classes = ["no or low anxiety", "moderate anxiety", "high anxiety"]
+        elif condition == "depression":
+            if "interest or pleasure" not in self.get_label(question + "_SQ001"):
+                raise ValueError("Question incompatible with specified condition type.")
+            base_score = 1
+            conversion = ["freq" for i in range(8)]
+            label = "depression"
+            classification_boundaries = [0, 4, 9, 14, 19, 24]
+            classes = [
+                "no to minimal depression",
+                "mild depression",
+                "moderate depression",
+                "moderately severe depression",
+                "severe depression",
+            ]
+        else:
```

Comment:
Can we make the `condition` an optional argument and try to infer it from the label of the question?
Something like:

```python
if condition is None:
    label = self.get_label(question)
    if "..." in label:
       condition = "state"
    elif "..." in label:
        condition = "trait"
    elif "..." in label:
        condition = "depression"
    else:
        raise ValueError(...)
```

Reply:
Implemented. Modified tests to make use of automatic inference.